web/src/routes/storage.tsx

```diff
@@ -38,46 +39,42 @@ const PATHS = {
   dasd: "/storage/dasd",
 };
 
-const routes = (): Route => {
-  const dasdRoute = {
-    path: PATHS.dasd,
-    element: <DASDPage />,
-    handle: { name: N_("DASD") },
-    loader: async () => probeDASD(),
-  };
-
-  const routes = {
-    path: PATHS.root,
-    handle: { name: N_("Storage"), icon: "hard_drive" },
-    children: [
-      {
-        index: true,
-        element: <ProposalPage />,
-      },
-      {
-        path: PATHS.targetDevice,
-        element: <DeviceSelection />,
-      },
-      {
-        path: PATHS.bootingPartition,
-        element: <BootSelection />,
-      },
-      {
-        path: PATHS.spacePolicy,
-        element: <SpacePolicySelection />,
+const routes = (): Route => ({
+  path: PATHS.root,
+  handle: { name: N_("Storage"), icon: "hard_drive" },
+  children: [
+    {
+      index: true,
+      element: <ProposalPage />,
+    },
+    {
+      path: PATHS.targetDevice,
+      element: <DeviceSelection />,
+    },
+    {
+      path: PATHS.bootingPartition,
+      element: <BootSelection />,
+    },
+    {
+      path: PATHS.spacePolicy,
+      element: <SpacePolicySelection />,
+    },
+    {
+      path: PATHS.iscsi,
+      element: <ISCSIPage />,
+      handle: { name: N_("iSCSI") },
+    },
+    {
+      path: PATHS.dasd,
+      element: <DASDPage />,
+      handle: { name: N_("DASD") },
+      loader: async () => {
+        if (!DASDSupported()) return redirect(PATHS.root);
```

Comment:
Not sure if worth mentioning that loader will not be executed until loading the route. Probably not.

Also, we should add a mechanism to avoid the application crashing because trying to go to the route before login **but in another round of routes improvements, not now** :wink:

BTW, thanks a lot for taking the time of doing manual testing of this part and catching such a bug.

Reply:
I can improve the message or add a note in the ZFCP PR just for not delay this one.